### PR TITLE
[new-app] easteregg/remove_deprecations

### PR DESCRIPTION
**Pull request by easteregg**

### CODE REVIEW

File: src/javascript/app_2/App/Components/Elements/Calendar/calendar.jsx [hunk 1]
```diff
@@ -9,10 +9,11 @@ class Calendar extends React.PureComponent {
     constructor(props) {
         super(props);
         const { date_format, start_date } = props;
-        const current_date = moment.utc(start_date).format(date_format);
+        const current_date = props.current_date || moment.utc(start_date).format(date_format);
+        const selected_date = props.selected_date || '';
         this.state = {
             calendar_date: current_date, // calendar date reference
-            selected_date: '',           // selected date
+            selected_date,           // selected date
```
[the remark this answers is not in the document]
Comment (easteregg) [reply src/javascript/app_2/App/Components/Elements/Calendar/calendar.jsx]: Thanks for pointing it out. I'll push a fix.

File: src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx [hunk 2]
```diff
@@ -5,22 +5,36 @@ import { IconArrow }   from 'Assets/Common';
...
+        if ((moment.utc(new_date, date_format).isValid() || !new_date)) {
+            if (!new_date) {
+                const current_date = moment.utc(start_date).format(date_format);
+                state.calendar_date = current_date;
+                state.selected_date = current_date;
+            } else {
+                state.selected_date = formatDate(new_date);
+                state.calendar_date = formatDate(new_date);
+            }
+        }
```
Comment (kellybinary) [inline src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx]: can we use `this.updateDatePickerValue(value, mode)` here as well?

File: src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx [hunk 3]
```diff
@@ -5,22 +5,36 @@ import { IconArrow }   from 'Assets/Common';
...
+        selected_date        : '',
     };
 
-    componentWillReceiveProps({ value, mode }) {
-        if (value === this.state.value) return;
-        this.updateDatePickerValue(value, mode);
+    static getDerivedStateFromProps({ value, mode, ...props }, state) {
+        const { date_format, start_date } = props;
+        const new_date = (mode === 'duration') ? moment.utc().add(value, 'days').format(date_format) : value;
+        if ((moment.utc(new_date, date_format).isValid() || !new_date)) {
```
Comment (kellybinary) [inline src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx]: nitpick: the parenthesis around the `if` condition seems unnecessary

File: src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx (hunk 2 above)
Comment (easteregg) [reply src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx]: We cannot use that, we do not have access to `this` in that method.

**Pull request merged**